### PR TITLE
🔎 Finegrained reward logging for GRPO

### PR DESCRIPTION
**Pull request by qgallouedec**
# What does this PR do?

When you've more than one reward func, you want to have the reward fo each one logged. Example:

```python
from datasets import load_dataset
from trl import GRPOConfig, GRPOTrainer

dataset = load_dataset("trl-lib/tldr", split="train")

# add a column ground_truth
dataset = dataset.map(lambda x: {"ground_truth": x["completion"]}, remove_columns=["completion"])

def my_verifier(completions, **kwargs):
    return [-len(completion) for completion in completions]

training_args = GRPOConfig(
    output_dir="Qwen2-0.5B-GRPO",
    # learning_rate=1e-5,
    logging_steps=2,
    gradient_accumulation_steps=2,
    max_prompt_length=64,
    max_completion_length=32,
    num_generations=4,
)
trainer = GRPOTrainer(
    model="Qwen/Qwen2-0.5B-Instruct",
    reward_funcs=[my_verifier, "trl-internal-testing/tiny-Qwen2ForSequenceClassification-2.5"],
    args=training_args,
    train_dataset=dataset,
)

trainer.train()
```

Screenshot 2025-01-25 at 00 52 20

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by plaguss — APPROVED** [review]

**Review by lewtun — APPROVED** [review]
Very nice improvement! LGTM
